core/Test/Tasty/Ingredients/ConsoleReporter.hs

```diff
@@ -194,46 +195,27 @@ consoleOutput :: (?colors :: Bool) => TestOutput -> StatusMap -> IO ()
 consoleOutput toutput smap =
   getTraversal . fst $ foldTestOutput foldTest foldHeading toutput smap
   where
-    foldTest _name printName getResult printResult =
+    foldTest _name printName getResult printResult opts =
       ( Traversal $ do
-          printName :: IO ()
+          let HideSuccesses hideSuccesses = lookupOption opts
           r <- getResult
-          printResult r
+          _ <- if resultSuccessful r && hideSuccesses
+               then do pure $ Any False
+               else do printName :: IO (); printResult r :: IO (); return $ Any True
```

Comment:
Any True and Any False don't do anything since the result is ignored. This hints at an issue: you are not erasing the empty headings like the original implementation did. Try compiling `core-tests/hide-successes-test.hs` on master and your branch and compare the output.